realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -0,0 +1,475 @@
+        activeTasks.add(task);
+        handler.post(task);
+    }
+
+    // Open both underlying Realms asynchronously. Once they are both ready, all tasks added in the meantime are
+    // started. If the Realms failed to open correctly any error will be reported through the `Callback.onError` callback.
+    private void openRealms() {
+        if (!openInProgress) {
+            openInProgress = true;
+            managementRealmOpenTask = Realm.getInstanceAsync(managementRealmConfig, new Realm.Callback() {
+                @Override
+                public void onSuccess(Realm realm) {
+                    managementRealm = realm;
+                    checkIfRealmsAreOpened();
+                }
+
+                @Override
+                public void onError(Throwable exception) {
+                    synchronized (errorLock) {
+                        managementRealmError = new ObjectServerError(ErrorCode.UNKNOWN, exception);
+                    }
+                    checkIfRealmsAreOpened();
+                }
+            });
+            permissionRealmOpenTask = Realm.getInstanceAsync(permissionRealmConfig, new Realm.Callback() {
+                @Override
+                public void onSuccess(Realm realm) {
+                    permissionRealm = realm;
+                    checkIfRealmsAreOpened();
+                }
+
+                @Override
+                public void onError(Throwable exception) {
+                    synchronized (errorLock) {
+                        permissionRealmError = new ObjectServerError(ErrorCode.UNKNOWN, exception);
+                    }
+                    checkIfRealmsAreOpened();
+                }
+            });
+        }
+    }
+
+    private void checkIfRealmsAreOpened() {
+        synchronized (errorLock) {
+            if ((permissionRealm != null || permissionRealmError != null)
+                && (managementRealm != null || managementRealmError != null)) {
+                openInProgress = false;
+                runDelayedTasks();
+            }
+        }
+    }
+
+    private void checkCallbackNotNull(Callback<?> callback) {
+        if (callback == null) {
+            throw new IllegalArgumentException("Non-null 'callback' required.");
+        }
+    }
+
+    private boolean isReady() {
+        return managementRealm != null && permissionRealm != null;
+    }
+
+    private void checkIfValidThread() {
+        // Checks if we are in thread that created the PermissionManager.
+        if (threadId != Thread.currentThread().getId()) {
+            throw new IllegalStateException("PermissionManager was accessed from the wrong thread. It can only be " +
+                                            "accessed on the thread it was created on.");
+        }
+    }
+
+    /**
+     * FIXME Add Javadoc
+     */
+    @Override
+    public void close() {
+        checkIfValidThread();
+        // If Realms are still being opened, abort that task
+        if (openInProgress) {
```

Comment:
Yup, but that boolean will not change unless both Realms are either opened or failed. See `checkIfRealmsAreOpened()`